source/dofs/dof_renumbering.cc

```diff
@@ -2442,6 +2442,74 @@ namespace DoFRenumbering
 
 
 
+  template <int dim>
+  void
+  lexicographic(DoFHandler<dim> &dof_handler, const double tolerance)
+  {
+    std::vector<types::global_dof_index> renumbering;
+    compute_lexicographic(renumbering, dof_handler, tolerance);
+    dof_handler.renumber_dofs(renumbering);
+  }
+
+
+  template <int dim>
+  void
+  compute_lexicographic(std::vector<types::global_dof_index> &new_dof_indices,
+                        const DoFHandler<dim>                &dof_handler,
+                        const double                          tolerance)
+  {
+    Assert(dynamic_cast<const parallel::distributed::Triangulation<dim> *>(
```

Comment:
I typically implement lexicographic numbering by computing a dof-wise downstream numbering https://dealii.org/developer/doxygen/deal.II/namespaceDoFRenumbering.html#a5175d4f7fb7256d3f5e21dcc3d722cb4 and using the downstream direction of `1, 1e-5, 1e-10`. This is not quite as tight as your selection, because of the tolerance. Would that be an alternative? I don't disagree with your implementation, but I'm just trying to see if we can reduce the amount of code we need to support (if the functionality is equivalent).

Reply:
WOW, I never thought about that, maybe because implementing the functionality was faster than thinking about a walkaround like that,  but I have to admit that's smart. We should definitely have lexicographic numbering in this namespace because I am not the only one who missed that. 
I don't know if that is a proper alternative; I would not feel right putting that walk-around in the library code. Although I will be fine with using it inside my code  (then I know what I did and I can blame myself if it causes bugs).

Reply:
I definitely agree that we should have a separate function, it is much more obvious to have that, so I am in full support for the interface. What I'm wondering is whether we could join the implementations (with the limitations in terms of roundoff I mentioned).

Reply:
> I typically implement lexicographic numbering by computing a dof-wise downstream numbering https://dealii.org/developer/doxygen/deal.II/namespaceDoFRenumbering.html#a5175d4f7fb7256d3f5e21dcc3d722cb4 and using the downstream direction of `1, 1e-5, 1e-10`. 

Oh, cool idea :-)

I agree that it would be nice to implement the current function in terms of the downstream function.

Reply:
Separately, you want the second and third entry in that vector to be of the order `1e-1 / diameter` and `1e-2 / diameter / diameter` for things to be correct even if the triangulation is geometrically very large.

Reply:
Regarding the tolerance topic, I think what @bangerth was trying to say is to use as the value for `tolerance` below a number related to
```c++
  const double actual_tolerance = tolerance * dof_handler.begin(0)->diameter();
```
or a related number. It is not going to be perfect, but probably good enough. One could also use `GridTools::diameter(dof_handler.get_triangulation());` to be on the safe side.